scripts/core/EXT_EventQueryKernelTimestamps.rst

```diff
@@ -0,0 +1,79 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_extension_event_query_kernel_timestamps:
+
+=========================================
+ Event Query Kernel Timestamps Extension
+=========================================
+
+API
+----
+
+* Functions
+
+
+    * ${x}EventQueryKernelTimestampsExt
+
+
+* Enumerations
+
+
+    * ${x}_event_query_kernel_timestamps_ext_version_t
+
+
+Event Query Kernel Timestamps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This extension enables the query of event timestamps to provide up to three views of the data, for a given event on a given device:
+
+1. Host timestamps, normalized with Device timestamps
```

Comment:
why return host timestamps normalized to device timestamps? I think the user feedback was that they would like to query for event timestamps that return a) data in time domain (rather than raw timestamps as we have now) and b) to return timestamps normalized to host time scale (as OCL does it with `clGetEventProfilingData`). So points 2 and 3 below seem to address the feedback. Also as noted in earlier commit, we already have `zeDeviceGetGlobalTimestamps` which would address 1)

Reply:
I think perhaps the way this was phrased is not quite as accurate or precise as it should be. The intention is a) and b) above. Only the final parameter is intended to provide the caller with raw (device) timestamps, associated with the given event.

Reply:
@aravindksg so you suggest we just returned deviceTimestamps, which are mapped to host. Host timestamps mapped to device are not needed? (would they be?)

Reply:
right, I'm not sure there's a customer need for host timestamps mapped to device time domain since the device operations are anyway host initiated. For host timings they could just as easily capture it themselves in upper levels. In this API, we mainly just need to provide device related info (but massage it per their feedback)

I wonder if there's a way we can define this new API with a `desc` which has a `pNext` such that if required we could extend the functionality of this API further based on customer need.

Reply:
if we extend this, would be with a properties, so instead of returning values in arguments, we could pass a props struct, and populate it, and use pNext for future extension of functionality.

@wdamon-intel : what do you think?